addons/html_editor/static/src/editor/core/movenode_plugin.js

```diff
@@ -0,0 +1,449 @@
+                        this._currentDropHint.style["width"] = `${currentDropHintSize}px`;
+                        dropzoneHintBox.append(this._currentDropHint);
+                        this._currentDropHintElementPosition = ["right", element];
+                    }
+                });
+                const removeDropHint = () => {
+                    if (this._currentDropHint) {
+                        this._currentDropHint.remove();
+                        this._currentDropHint = null;
+                    }
+                    this._currentDropHintCommand = null;
+                };
+                dropzoneBox.addEventListener("mouseleave", removeDropHint);
+            }
+
+            this._dropzonesContainer.append(dropzoneBox);
+            this._dropzoneHintContainer.append(dropzoneHintBox);
+        }
+    }
+    _stopDropzones(movableElement) {
+        this.editable.classList.remove("oe-editor-dragging");
+        this._dropzonesContainer.replaceChildren();
+        this._dropzoneHintContainer.replaceChildren();
+
+        if (this._currentDropHintElementPosition) {
+            const [position, focusElelement] = this._currentDropHintElementPosition;
+            this._currentDropHintElementPosition = undefined;
+            const previousParent = movableElement.parentElement;
+            if (position === "top") {
+                focusElelement.before(movableElement);
+            } else if (position === "bottom") {
+                focusElelement.after(movableElement);
+            }
+            if (previousParent.innerHTML.trim() === "") {
+                const p = document.createElement("p");
+                const br = document.createElement("br");
+                p.append(br);
+                previousParent.append(p);
+            }
+            this.shared.setSelection({ anchorNode: movableElement, anchorOffset: 0 });
+            this.dispatch("ADD_STEP");
+        }
+    }
+    _onMousemove(e) {
+        this._updateAnchorWidgets(e.target);
+    }
+    _onDocumentKeydown() {
+        // Hide the move widget upon keystroke for visual clarity and provide
+        // visibility to a collaborative avatar.
+        this._removeMoveWidget();
+    }
+    _onDocumentMousemove(e) {
+        if (this._resetHooksNextMousemove) {
+            this._resetHooksNextMousemove = false;
+            this._removeMoveWidget();
+            this._updateHooks();
+        }
+        if (this._editableRect && !isPointInside(this._editableRect, e.clientX, e.clientY)) {
+            this._removeMoveWidget();
+        }
+    }
+}
+
+function isNodeMovable(node) {
+    return (
+        node.parentElement?.getAttribute("contentEditable") === "true" &&
+        !node.parentElement.closest(".o_editor_banner")
+    );
+}
+
+function isPointInside(rect, x, y) {
+    return rect.left <= x && rect.right >= x && rect.top <= y && rect.bottom >= y;
+}
+
+const simpleDraggableHook = {
```

Comment:
Why do you prepend this variable name with "simple"? Are there any other draggable hooks?

Reply:
these lines were not created by me originally